app/services/cancellation_fee_calculator.rb

```diff
@@ -12,10 +12,10 @@ def fee
 
   def calculated_fee
     unless defined?(@calculated_fee)
-      original_canceled_at = @reservation.canceled_at
-      @reservation.canceled_at = Time.current
+      original_canceled_at = @reservation.order_detail.canceled_at
```

Comment:
Really just that this is now clunkly and getting close to law of demeter territory. Might be we don't worry about this for now. A better solution might be to have the cancellation fee calculator work on an OrderDetail rather than a reservation (I could see this coming up when we get to multiple signup).